compiler/src/iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -475,8 +485,9 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
   // example, the LLVM AOT backend may combine all executable targets for the
   // same architecture into a single executable and link it as a shared
   // library.
-  if (transformOptions.linkExecutables) {
-    passManager.addPass(IREE::HAL::createLinkExecutablesPass({targetRegistry}));
+  if (transformOptions.linkExecutables && clLinkExecutables) {
```

Comment:
It is strange to me that this is `&&` but if it isn't then to disable linking we'd have to plumb through another flag for the transform option and disable that too. I think it's fine because linking should pretty much always be on but if there are any other suggestions.